### ISSUE TITLE
Does not look for config file in $HOME/.config/alacritty/alacritty.toml as described in man page

### ISSUE THREAD

**Issue by doc1623** [alacritty/alacritty#7366]
I was trying to use #3 from the man page as configuration location. My $XDG_CONFIG_HOME isn't set. #4 works, but not #3. I am using symlinks, but #4 works with the same target. 

### System

OS: Linux 
```
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.3 LTS"
```

Version: alacritty 0.13.0-dev (3ffd6c8f)
Linux: X11

### Logs
`[0.000143872s] [INFO ] [alacritty_config_derive] No config file found; using default`

From Man Page
```
           Alacritty looks for the configuration file at the following paths:
               1.   $XDG_CONFIG_HOME/alacritty/alacritty.toml
               2.   $XDG_CONFIG_HOME/alacritty.toml
               3.   $HOME/.config/alacritty/alacritty.toml
               4.   $HOME/.alacritty.toml
```

```
[3630 23-11-13 17:25]%echo $HOME/.config/alacritty/alacritty.toml
/home/me/.config/alacritty/alacritty.toml
[3630 23-11-13 17:25]%ls -larth $HOME/.config/alacritty/alacritty.toml
lrwxrwxrwx 1 me me 46 Nov 13 17:19 /home/me/.config/alacritty/alacritty.toml ->
 git/configs/terminals/alacritty/alacritty.toml
```

See that it doesn't work until I create the symlink for the #4 location and then you see that the theme import works from the config
```
[3630 23-11-13 17:30]%alacritty -vvv|grep -i config
[0.000157953s] [INFO ] [alacritty_config_derive] No config file found; using default
[0.008917606s] [DEBUG] [alacritty] Picked GL Config:
                                     config_api: Api(OPENGL | GLES1 | GLES2)
[3630 23-11-13 17:31]%ln -s git/configs/terminals/alacritty/alacritty.t
oml $HOME/.alacritty.toml
[3630 23-11-13 17:32]%alacritty -vvv|grep -i config
[0.000260205s] [INFO ] [alacritty] Configuration files loaded from:
                                     "/home/me/.config/alacritty/alacritty-theme/themes/solarized_dark.toml"
[0.009585935s] [DEBUG] [alacritty] Picked GL Config:
                                     config_api: Api(OPENGL | GLES1 | GLES2)
```

**Comment by kchibisov** [comment]
Could you try symlinking `$HOME/.config/alacritty/alacritty.toml` to an absolute path instead of relative?

**Comment by aadi58002** [comment]
I tried using the `alacritty.toml` under `$XDG_CONFIG_HOME/alacritty/alacritty.toml` but it won't detect it no matter what i tried in the end only the `$XDG_CONFIG_HOME/alacritty/alacritty.yml` file worked for me and i think it should be mentioned in the README that toml is being migrated too and you should try a yml config if you are having issues.

**Comment by kchibisov** [comment]
>I tried using the alacritty.toml under $XDG_CONFIG_HOME/alacritty/alacritty.toml but it won't detect it no matter what i tried in the end only the $XDG_CONFIG_HOME/alacritty/alacritty.yml file worked for me and i think it should be mentioned in the README that toml is being migrated too and you should try a yml config if you are having issues.

unless you've build version yourself from master you should use `yml`.

**Comment by aadi58002** [comment]
I think this should be mentioned in the readme to remove confusion

**Comment by kchibisov** [comment]
![image](https://github.com/alacritty/alacritty/assets/27620401/9da870dd-89ac-4aba-87c1-baabc8abca70)

**Comment by aadi58002** [comment]
> ![image](https://user-images.githubusercontent.com/27620401/282708116-9da870dd-89ac-4aba-87c1-baabc8abca70.png)

Sorry I didn't notice it.

**Comment by doc1623** [comment]
Just as an fyi @kchibisov, I had tried the absolute path. I really only recreating it using the $HOME variable, so as not to give my name, in the example. I see this got closed by @chrisduerr, was no one able to reproduce? Is that what "needs a repro" means? 

Did I document it ok? I can do a bit more, but unless someone else has the same issue, I guess it might be something weird on my side? I have it working with the symlink as the direct dot file under $HOME... just thought to help as I was using the dev build.

**Comment by chrisduerr** [comment]
Oh sorry I didn't notice all the comments were by a different person.

> I have it working with the symlink as the direct dot file under $HOME

So the issue is fixed then? Did you actually create the symlink itself as an invalid relative path?

**Comment by doc1623** [comment]
To my knowledge, the issue isn't fixed. I just used the #4 instead of #3 from the manual... just the /home/me/.alacritty.toml instead of $HOME/.config/alacritty/alacritty.toml

**Comment by doc1623** [comment]
no worries, again, I was just thinking to help. As I'm using it :)

**Comment by kchibisov** [comment]
@doc1623 the only way you can have such a thing is when your symlink is dead, that's why I suggested to create an absolute one. Because they are relative they could change based on the place they are.

**Comment by doc1623** [comment]
@kchibisov I believe, I tried without a symlink too, just file, but I can try again. 

I'm not exactly sure what you're saying "have such a thing"?  Like I said, I only recreated with the $HOME variable for this example. I tried before with the full path e.g.  /home/me/.config/alacritty/alacritty.toml  many times or are you suggesting a hard link?  I didn't have a little issue and go straight here. It's not huge on my end, but before I wrote it up, I tried a bunch of times and ways. I'll try no symlink at all...just to give it a shot (I think, I already did, but I'll do it again)

**Comment by kchibisov** [comment]
I'm saying that you either did it wrong with symlinks. You can just place a file there, but we do look into `$HOME` variable to build the path for 3. and 4.

**Comment by doc1623** [comment]
It will take a symlink for #3 but not for #4. 

In other words,
`ln -s target $HOME/.alacritty.toml` works, but `ln -s target $HOME/.config/alacritty/alacritty.toml` does not.  

I did try the file directly from the latter, and it does, so, it's only when using the symlink that the config directory doesn't work. So it seems someone is checking for symlinks when looking under the config directory. I copied the screen to a file and then did a sed substitute for my name with me.... so the only mod done for the below. 







First line is the last of a `ls -larth` to show the working link that I break with the remove. 
```
lrwxrwxrwx   1 me me   46 Nov 13 17:32 .alacritty.toml -> git/configs/terminals/alacritty/alacritty.toml
[3630 23-11-14 13:24]me@/home/me%rm .alacritty.toml
[3630 23-11-14 13:25]me@/home/me%alacritty -vvv|grep -i config
[0.000152279s] [INFO ] [alacritty_config_derive] No config file found; using default
[0.009635769s] [DEBUG] [alacritty] Picked GL Config:
                                     config_api: Api(OPENGL | GLES1 | GLES2)
[3630 23-11-14 13:25]me@/home/me%cp git/configs/terminals/alacritty/alacritty.toml .config/alacritty
cp: not writing through dangling symlink '.config/alacritty/alacritty.toml'
[3630 23-11-14 13:25]me@/home/me%rm .config/alacritty/alacritty.toml
[3630 23-11-14 13:27]me@/home/me%cp git/configs/terminals/alacritty/alacritty.toml .config/alacritty
[3630 23-11-14 13:27]me@/home/me%alacritty -vvv|grep -i config
[0.000272071s] [INFO ] [alacritty] Configuration files loaded from:
                                     "/home/me/.config/alacritty/alacritty.toml"
                                     "/home/me/.config/alacritty/alacritty-theme/themes/solarized_dark.toml"
[0.010083041s] [DEBUG] [alacritty] Picked GL Config:
                                     config_api: Api(OPENGL | GLES1 | GLES2)
[0.242123929s] [TRACE] [alacritty_terminal] Configuring charset G0 as Ascii
[0.248615375s] [TRACE] [alacritty_terminal] Configuring charset G0 as Ascii
[0.248630520s] [TRACE] [alacritty_terminal] Configuring charset G0 as Ascii
[0.248661633s] [TRACE] [alacritty_terminal] Configuring charset G0 as Ascii
[0.252084410s] [TRACE] [alacritty_terminal] Configuring charset G0 as Ascii
[0.252099027s] [TRACE] [alacritty_terminal] Configuring charset G0 as Ascii
[0.253717217s] [TRACE] [alacritty_terminal] Configuring charset G0 as Ascii
[0.254497217s] [TRACE] [alacritty_terminal] Configuring charset G0 as Ascii
[0.254512027s] [TRACE] [alacritty_terminal] Configuring charset G0 as Ascii
[6.580314708s] [TRACE] [alacritty_terminal] Configuring charset G0 as Ascii
[3630 23-11-14 13:27]me@/home/me%rm .config/alacritty/alacritty.toml
[3630 23-11-14 13:28]me@/home/me%ln -s git/configs/terminals/alacritty/alacritty.toml .config/alacritty/alacritty.toml
[3630 23-11-14 13:29]me@/home/me%ls -l .config/alacritty/alacritty.toml
lrwxrwxrwx 1 me me 46 Nov 14 13:29 .config/alacritty/alacritty.toml -> git/configs/terminals/alacritty/alacritty.toml
[3630 23-11-14 13:30]me@/home/me%alacritty -vvv|grep -i config
[0.000169042s] [INFO ] [alacritty_config_derive] No config file found; using default
[0.009857339s] [DEBUG] [alacritty] Picked GL Config:
                                     config_api: Api(OPENGL | GLES1 | GLES2)
[3630 23-11-14 13:30]me@/home/me%ln -s git/configs/terminals/alacritty/alacritty.toml $HOME/.alacritty.
toml
[3630 23-11-14 13:31]me@/home/me%alacritty -vvv|grep -i config
[0.000271611s] [INFO ] [alacritty] Configuration files loaded from:
                                     "/home/me/.config/alacritty/alacritty-theme/themes/solarized_dark.toml"
[0.009756747s] [DEBUG] [alacritty] Picked GL Config:
                                     config_api: Api(OPENGL | GLES1 | GLES2)
[0.328588319s] [TRACE] [alacritty_terminal] Configuring charset G0 as Ascii
[0.329753967s] [TRACE] [alacritty_terminal] Configuring charset G0 as Ascii
[0.329799775s] [TRACE] [alacritty_terminal] Configuring charset G0 as Ascii
[0.329846779s] [TRACE] [alacritty_terminal] Configuring charset G0 as Ascii
[0.330484280s] [TRACE] [alacritty_terminal] Configuring charset G0 as Ascii
[0.330498432s] [TRACE] [alacritty_terminal] Configuring charset G0 as Ascii
```

**Comment by kchibisov** [comment]
Please, do absolute symlink.

`ln -s /home/me/git/configs/terminals/alacritty/alacritty.toml /home/me/.config/alacritty/alacritty.toml`.

**Comment by chrisduerr** [comment]
> First line is the last of a ls -larth to show the working link that I break with the remove.

Running `ls` does not prove the link actually works. Try `cat` instead.

**Comment by doc1623** [comment]
ok, let me try again with the absolute.. and tail or something (just to keep the output small)

**Comment by doc1623** [comment]
Well, I don't understand it, but the absolute path works. @kchibisov you said things can change, but the link doesn't work from the get go... that's new to me. Thank YOU BOTH. I've learned something from now on absolute path symlinks and test with head -1 or something that reads... 

Sorry, if this was a waste of time. I thought I was being consistent in how I did both symlinks... so I really don't know. Regardless, thanks again
